internal/pkg/cli/env_package.go

```diff
@@ -0,0 +1,298 @@
+		if err != nil {
+			return fmt.Errorf("upload assets for environment %q: %v", o.envName, err)
+		}
+	}
+	res, err := deployer.GenerateCloudFormationTemplate(&deploy.DeployEnvironmentInput{
+		RootUserARN:         principal.RootUserARN,
+		CustomResourcesURLs: urls,
+		Manifest:            mft,
+	})
+	if err != nil {
+		return fmt.Errorf("generate CloudFormation template from environment %q manifest: %v", o.envName, err)
+	}
+	if err := o.setWriters(); err != nil {
+		return err
+	}
+	if err := o.write(o.tplWriter, res.Template); err != nil {
+		return err
+	}
+	return o.write(o.paramsWriter, res.Parameters)
+}
+
+func (o *packageEnvOpts) getAppCfg() (*config.Application, error) {
+	if o.appCfg != nil {
+		return o.appCfg, nil
+	}
+	cfg, err := o.cfgStore.GetApplication(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.appCfg = cfg
+	return o.appCfg, nil
+}
+
+func (o *packageEnvOpts) getEnvCfg() (*config.Environment, error) {
+	if o.envCfg != nil {
+		return o.envCfg, nil
+	}
+	cfg, err := o.cfgStore.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %q in application %q: %w", o.envName, o.appName, err)
+	}
+	o.envCfg = cfg
+	return o.envCfg, nil
+}
+
+func (o *packageEnvOpts) validateOrAskEnvName() error {
+	if o.envName != "" {
+		if _, err := o.getEnvCfg(); err != nil {
+			log.Errorf("It seems like environment %s is not added in application %s yet. Have you run %s?\n",
+				o.envName, o.appName, color.HighlightCode("copilot env init"))
+			return err
+		}
+		return nil
+	}
+
+	name, err := o.sel.LocalEnvironment("Select an environment manifest from your workspace", "")
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = name
+	return nil
+}
+
+func (o *packageEnvOpts) environmentManifest() (*manifest.Environment, error) {
+	envCfg, err := o.getEnvCfg()
+	if err != nil {
+		return nil, err
+	}
+	raw, err := o.ws.ReadEnvironmentManifest(envCfg.Name)
+	if err != nil {
+		return nil, fmt.Errorf("read manifest for environment %q: %w", envCfg.Name, err)
+	}
+	interpolated, err := o.newInterpolator(o.appName, o.envName).Interpolate(string(raw))
+	if err != nil {
+		return nil, fmt.Errorf("interpolate environment variables for %q manifest: %w", o.envName, err)
+	}
+	mft, err := manifest.UnmarshalEnvironment([]byte(interpolated))
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal environment manifest for %q: %w", envCfg.Name, err)
+	}
+	if err := mft.Validate(); err != nil {
+		return nil, fmt.Errorf("validate environment manifest for %q: %w", envCfg.Name, err)
+	}
+	return mft, nil
+}
+
+func (o *packageEnvOpts) setWriters() error {
+	if o.outputDir == "" {
+		return nil
+	}
+	if err := o.fs.MkdirAll(o.outputDir, 0755); err != nil {
+		return fmt.Errorf("create directory %q: %w", o.outputDir, err)
+	}
+
+	path := filepath.Join(o.outputDir, fmt.Sprintf("env-%s.yml", o.envName))
```

Comment:
An extreme edge case would be having a workload called "env". Running `svc package --name env --env test` would write the output to `env-test.params.json`  - because it's formatted as `<svc>-<env>.params.json` - which collides with this naming scheme. 

I don't think we need to worry about this too much though but just to call it out!

Reply:
then it may have 
```
/infrastructure
   /svc-env.yaml
   /env_test.yaml
```
The inconsistency on the usage of `-` and `_` is a bit bothering 😓

Reply:
Right right but it seems like we need to choose a poison

Reply:
what do you think of 
`./infrastructure/environments/env-test.yaml`

we can then generate the pipeline manifest with this path in mind

Reply:
yea but then the output dir isn't really the one they specify, which is not consistent with the workload behavior. And could end up being `./infra/env/environments/env-test.yaml` if they specify the output dir to be `./infra/env/`. I would prefer to leave as is...

Reply:
Renamed the file format to `%s.env.yml` and `%s.env.params.json`